pallets/eth-bridge/src/lib.rs

```diff
@@ -801,15 +813,25 @@ pub mod pallet {
             match ValidEvents::try_from(&discovered_event.event.event_id.signature) {
                 Some(valid_event) =>
                     if active_range.event_types_filter.contains(&valid_event) {
-                        process_ethereum_event::<T>(&discovered_event.event);
+                        match process_ethereum_event::<T>(&discovered_event.event) {
+                            Ok(_) => {},
+                            Err(_) => {
+                                log::error!("💔 Duplicate Event Submission");
+                                <Pallet<T>>::deposit_event(Event::<T>::DuplicateEventSubmission {
+                                    eth_event_id: discovered_event.event.event_id.clone(),
+                                });
+                            },
+                        }
```

Comment:
```suggestion
                        if process_ethereum_event::<T>(&discovered_event.event).is_err() {
                                log::error!("💔 Duplicate Event Submission");
                                <Pallet<T>>::deposit_event(Event::<T>::DuplicateEventSubmission {
                                    eth_event_id: discovered_event.event.event_id.clone(),
                                });
                        }
```

Reply:
Maybe we should have a generic `InvalidEventSubmission` instead of `DuplicateEventSubmission`? We can then use this for 0 transaction hashes too (not just existing events)